firebase_admin/_messaging_encoder.py

```diff
@@ -0,0 +1,700 @@
+            'light_settings': cls.encode_light_settings(notification.light_settings),
+            'visibility': _Validators.check_string(
+                'AndroidNotification.visibility', notification.visibility, non_empty=True),
+            'notification_count': _Validators.check_number(
+                'AndroidNotification.notification_count', notification.notification_count)
+        }
+        result = cls.remove_null_values(result)
+        color = result.get('color')
+        if color and not re.match(r'^#[0-9a-fA-F]{6}$', color):
+            raise ValueError(
+                'AndroidNotification.color must be in the form #RRGGBB.')
+        if result.get('body_loc_args') and not result.get('body_loc_key'):
+            raise ValueError(
+                'AndroidNotification.body_loc_key is required when specifying body_loc_args.')
+        if result.get('title_loc_args') and not result.get('title_loc_key'):
+            raise ValueError(
+                'AndroidNotification.title_loc_key is required when specifying title_loc_args.')
+
+        event_time = result.get('event_time')
+        if event_time:
+            result['event_time'] = str(event_time.isoformat()) + 'Z'
+
+        priority = result.get('notification_priority')
+        if priority and priority not in ('min', 'low', 'default', 'high', 'max'):
+            raise ValueError('AndroidNotification.priority must be "default", "min", "low", "high" '
+                             'or "max".')
+        if priority:
+            result['notification_priority'] = 'PRIORITY_' + priority.upper()
+
+        visibility = result.get('visibility')
+        if visibility and visibility not in ('private', 'public', 'secret'):
+            raise ValueError(
+                'AndroidNotification.visibility must be "private", "public" or "secret".')
+        if visibility:
+            result['visibility'] = visibility.upper()
+
+        vibrate_timings_millis = result.get('vibrate_timings')
+        if vibrate_timings_millis:
+            vibrate_timings_secs = []
+            for msec in vibrate_timings_millis:
+                formated_string = cls.encode_milliseconds(
+                    'AndroidNotification.vibrate_timings_millis', msec)
+                vibrate_timings_secs.append(formated_string)
+            result['vibrate_timings'] = vibrate_timings_secs
+        return result
+
+    @classmethod
+    def encode_light_settings(cls, light_settings):
+        """Encodes a LightSettings instance into JSON."""
+        if light_settings is None:
+            return None
+        if not isinstance(light_settings, _messaging_utils.LightSettings):
+            raise ValueError(
+                'AndroidNotification.light_settings must be an instance of LightSettings class.')
+        result = {
+            'color': _Validators.check_string(
+                'LightSettings.color', light_settings.color, non_empty=True),
+            'light_on_duration': cls.encode_milliseconds(
```

Comment:
This will implicitly also accept `timedelta` typed arguments. Was that our intention or just an unintended consequence? Not a huge deal either way, just wanted to make sure.

Reply:
This was an unintended consequence. We do the same thing for `ttl`, too. I thought it would be nice to accept both `number` and `timedelta` in the utility function. If we want to restrict `light_on/off_duration` to `number`, we can add a validation step prior to encoding milliseconds  and throw if it is not a number.

Reply:
I think this is fine. But let's just document it as a milliseconds API for simplicity.